conference.js

```diff
@@ -1977,35 +1977,42 @@ export default {
      * @private
      */
     _initDeviceList() {
-        if (JitsiMeetJS.mediaDevices.isDeviceListAvailable() &&
-            JitsiMeetJS.mediaDevices.isDeviceChangeAvailable()) {
-            JitsiMeetJS.mediaDevices.enumerateDevices(devices => {
-                // Ugly way to synchronize real device IDs with local
-                // storage and settings menu. This is a workaround until
-                // getConstraints() method will be implemented in browsers.
-                if (localAudio) {
-                    APP.settings.setMicDeviceId(
-                        localAudio.getDeviceId(), false);
-                }
+        JitsiMeetJS.mediaDevices.isDeviceListAvailable()
+            .then(isDeviceListAvailable => {
+                if (isDeviceListAvailable
+                        && JitsiMeetJS.mediaDevices.isDeviceChangeAvailable()) {
```

Comment:
Style thought: What about returning early if falsy instead of nesting the logic in the if?